### DISCUSSION TITLE
srsUE connects to srsgNB with failure: Proc "Cell Selection" - Completed with failure.

### DISCUSSION THREAD

**Discussion by WanHaoRan** [srsran/srsRAN_Project#140] [General Help]
Hi, I am trying to reproduce the [application note ](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html) but I cannot connect my srsUE to the srsgNB. I am using two B210 for gNB and UE. I am using the "performance" mode for both laptop computer.

The gNB seems to work fine, but the UE cannot connect to the gNB. The UE reports Proc "Cell Selection" - Completed with failure in its log. It seems that the UE doesn't see the network at all. 

Here are my config files:
UE conf: [ue_rf.conf.txt](https://github.com/srsran/srsRAN_Project/files/12140088/ue_rf.conf.txt)
gNB conf: [gnb_fdd_srsue.yaml.txt](https://github.com/srsran/srsRAN_Project/files/12140108/gnb_fdd_srsue.yaml.txt)

Log files:
UE log: [ue.log](https://github.com/srsran/srsRAN_Project/files/12140111/ue.log)
gNB log: [gnb.log](https://github.com/srsran/srsRAN_Project/files/12140109/gnb.log)
gNB ngap.pcap: [gnb_ngap.pcap.txt](https://github.com/srsran/srsRAN_Project/files/12140113/gnb_ngap.pcap.txt)
gNB mac.pcap: [gnb_mac.pcap.txt](https://github.com/srsran/srsRAN_Project/files/12140114/gnb_mac.pcap.txt)

Terminal outputs:
UE: [ue_terminal_output.txt](https://github.com/srsran/srsRAN_Project/files/12140115/ue_terminal_output.txt)
gNB: [gnb_terminal_output.txt](https://github.com/srsran/srsRAN_Project/files/12140117/gnb_terminal_output.txt)

Open5gs settings:
amf: [amf.yaml.txt](https://github.com/srsran/srsRAN_Project/files/12140128/amf.yaml.txt)
upf: [upf.yaml.txt](https://github.com/srsran/srsRAN_Project/files/12140129/upf.yaml.txt)

Open5gs subscriber setting:
![Screenshot from 2023-07-23 18-23-28](https://github.com/srsran/srsRAN_Project/assets/26237558/500b7c20-4580-455d-821f-791a71e5a825)

Could you tell me how to locate the mistakes and how to fix it?

Also, could you tell me about the meanings of "overflow", "underflow" and "late"? The gNB reports underflow and late message sometimes and UE reports overflow.

Thank you very much in advance!

**Comment by WanHaoRan** [comment]
Some updates:

I used a X310 as gNB instead of B210, then I got the same issue as [this discuss](https://github.com/srsran/srsRAN_Project/discussions/112). The previous problem could be caused by the clock of B210, but I am not very sure about it.

**Comment by WanHaoRan** [comment]
Some updates:

I tweaked some configuration for this scenario, mainly the rx_gain and rf_cfo to make it work successfully. I tuned B210's receving gain up to 70dB so it can detect the cell and with the CFO feedback in ue.log I set the CFO manually and it can synchronize successfully. Then the gNB returned that the TA of Preambles is not in some timing window, I shortened the  "time_adv_nsamples" in UE's configuration file and it worked. 

In summary, used X310 as gNB and B210 as UE. With config files: [config_for_gNB_UE.zip](https://github.com/srsran/srsRAN_Project/files/12186591/config_for_gNB_UE.zip). I also tried to use other combinations such as B210/X310 as gNB, and B210/X310 as UE but X310 as gNB and B210 as UE is the only combination that works. Also, I am not using the external clocks.

Now the RRC is connected, and I will proceed and update.

**Reply by rajugbv** [reply]
HI 
how did you find out your CFO feedback valuein ue log file .when i tried with -5200 RRC can connect for few seconds
i have attached my ue file can you please check it
 
[ue.log](https://github.com/srsran/srsRAN_Project/files/12410560/ue.log)

**Reply by WanHaoRan** [reply]
Hello,
It's in the first few lines of the ue.log, in your file it's "2023-07-24T17:37:02.453050 [RRC-NR ] [I] Proc "Cell Selection" - Cell search found ARFCN=0 PCI=1 epre=-29.9 snr=+21.7 cfo=+4759.6 delay=+0.0  sfn=991 ssb_idx=0 hrf=n scs=15 ssb_offset=6 dmrs_typeA_pos=pos2 coreset0=8 ss0=0 barred=n intra_freq_reselection=n spare=0" and you can see the cfo is +4759.6.

**Comment by WanHaoRan** [comment]
Some updates:

With a GPSO and put the devices closer, I can connect the srsUE (B210) to the srsgNB (X310). But the connection is not stable, it stays for a while, like tens of seconds, shortened by iperf3 test. Probably limited by the CPU's performance? I am using two laptops with i5-7300.

Another problem is that I tried to use X310 as UE, with the same laptops, it cannot see the cell. But with B210 it's good. Does anybody have some clues about this issue? CPU issue again?

**Comment by WanHaoRan** [comment]
I manage to connect the COTS phone (Moto Stylus 5G) to the network, but it seems that I have to use the Verizon's MCC (311) & MNC (484) setting. Because I can use my comercial SIM card to see the network, but with "00101" the mobile phone seems to block the 5G mode. So I set the SIM card's MCC and MNC accordingly, it works.

**Comment by WanHaoRan** [comment]
Since it worked out, I will close this discussion. If a better CPU can help to solve "underflow" and disconnecting problem, I'll come back to update.
